snapshot/snapshotfs/upload.go

```diff
@@ -255,29 +255,52 @@ func (u *Uploader) uploadDir(ctx context.Context, rootDir fs.Directory, policyTr
 	return de, err
 }
 
-func (u *Uploader) foreachEntryUnlessCancelled(relativePath string, entries fs.Entries, cb func(entry fs.Entry, entryRelativePath string) error) error {
-	for _, entry := range entries {
-		if u.IsCancelled() {
-			return errCancelled
-		}
+func (u *Uploader) foreachEntryUnlessCancelled(ctx context.Context, parallel int, relativePath string, entries fs.Entries, cb func(ctx context.Context, index int, entry fs.Entry, entryRelativePath string) error) error {
+	type entryWithIndex struct {
```

Comment:
Given that the entries will be sorted to create the corresponding snapshot directory, would it make sense to send the results back through a channel instead of passing an index for the entry? This is assuming the result type is always the same.

Reply:
implemented this suggestion, much cleaner!